src/Aspose.Imaging.Cloud.Sdk.Test/Api/DeskewApiTests.cs

```diff
@@ -0,0 +1,198 @@
+            { FileFormat.Djvu, new string[] {"djvu"} },
+            { FileFormat.Dng, new string[] {"dng"} },
+            { FileFormat.Emf, new string[] {"emf"} },
+            { FileFormat.Gif, new string[] {"gif"} },
+            { FileFormat.Jpeg, new string[] {"jpeg", "jpg"} },
+            { FileFormat.Jpeg2000, new string[] {"j2k", "jp2", "jpm", "jpx"} },
+            { FileFormat.Png, new string[] {"png"} },
+            { FileFormat.Psd, new string[] {"psd"} },
+            { FileFormat.Tiff, new string[] {"tiff", "tif"} },
+            { FileFormat.Webp, new string[] {"webp"} },
+            { FileFormat.Wmf, new string[] {"wmf"} }
+        };
+
+
+        /// <summary>
+        /// Deskew image test
+        /// </summary>
+        /// <param name="formatExtension">image format to test</param>
+        /// <param name="resizeProportionally">Resize proportionally</param>
+        /// <param name="bkColor">Background color</param>
+        [TestCase("jpg", true)]
+        [TestCase("jpg", false)]
+        [TestCase("jpg", false, "green")]
+#if EXTENDED_TEST
+        [TestCase("bmp", true)]
+        //todo: enable after save of images of these fomats is implemented
+        //[TestCase("dicom", true)]
+        //[TestCase("djvu", true)]
+        //[TestCase("dng", true)]
+        // TODO: enable after IMAGINGCLOUD-51 is resolved
+        //[TestCase("gif", true)]
+        [TestCase("j2k", true)]
+        [TestCase("jp2", true)]
+        [TestCase("jpx", true)]
+        [TestCase("jpm", true)]
+        [TestCase("jpeg", true)]
+        [TestCase("png", true)]
+        [TestCase("psd", true)]
+        [TestCase("tiff", true)]
+        [TestCase("tif", true)]
+        [TestCase("webp", true)]
+#endif
+        public void DeskewImageTest(string formatExtension, bool resizeProportionally, string bkColor = null)
+        {
+            foreach (StorageFile inputFile in InputTestFiles.Where(f => f.Name.EndsWith("." + formatExtension)))
+            {
+                this.TestGetRequest(
+                    "DeskewImageTest",
+                    $"Input image: {inputFile.Name}; Output format: {formatExtension}; Resize proportionally: {resizeProportionally}; Background color: {bkColor ?? "null"};",
+                    inputFile.Name,
+                    delegate
+                    {
+                        var request = new DeskewImageRequest(inputFile.Name, resizeProportionally, bkColor, TempFolder,
+                            TestStorage);
+                        return ImagingApi.DeskewImage(request);
+                    },
+                    delegate(ImagingResponse originalProperties, ImagingResponse resultProperties, Stream resultStream)
+                    {
+                        Assert.NotNull(resultStream);
+                        resultStream.Position = 0;
+                        Assert.IsTrue(ImageFormatsEqual(Image.Load(resultStream).FileFormat, formatExtension));
```

Comment:
@dmitrypasyutin Is it really necessary to check format extensions for that test case? Could you please tell, what benefits does this check provide to demand to include Aspose.Imaging package reference to the project?

Reply:
@s-zubov Aspose.Imaging.Cloud implements several similar methods (Crop, Resize, Update) that allow to modify an image and save it in a different format, which is set via the "format" parameter. Method Deskew differs from them in that it doesn't allow format change and must return an image in the original format only, so I decided to check that feature using that test. I used Aspose.Imaging library to extract the format of the resulting image because previously it was recommended to me to use Aspose libraries to work with images

Reply:
@dmitrypasyutin Isn't it a bit too complicated for SDK tests? If you think that this check is required, consider implementing it in other SDKs as well.

Reply:
@s-zubov I agree, that's too complicated, and should be moved to integration tests